tools/git-utils.sh

```diff
@@ -116,3 +116,26 @@ unpack_from_cache() {
     message "UNPACK" "${SUBDIR}  [ref: $1]"
     git_cache archive "$1" "${SUBDIR}" | tar -x --touch "${SUBDIR}"
 }
+
+# This stores a .tar file from stdin into the cache as a tree object.
+# Returns the ID.  Opposite of `git archive`, basically.
+tar_to_cache() {
+    init_cache
+
+    # Use a sub-shell to enable cleanup of the temporary directory
+    (
+        tmpdir="$(mktemp --tmpdir --directory cockpit-tar-to-git.XXXXXX)"
+        trap "rm -r '${tmpdir}'" EXIT
```

Comment:
I'd recommend to add `INT QUITE PIPE` as well (these four are my standard invocation)

Reply:
So I tested this by (for example) pressing `^C` and the `EXIT` hook still ran...

I guess that's because the sigint was actually being delivered to the subcommand and not the shell itself...